subsys/net/lib/openthread/platform/flash.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <flash.h>
+
+#include "platform-zephyr.h"
+
+static struct device *flash_dev;
+static size_t ot_flash_size;
+static size_t ot_flash_offset;
+
+static inline u32_t mapAddress(u32_t aAddress)
+{
+	return aAddress + ot_flash_offset;
+}
+
+otError utilsFlashInit(void)
+{
+	int i;
+	struct flash_pages_info info;
+	size_t pages_count;
+
+	flash_dev = device_get_binding(
+			CONFIG_OT_PLAT_FLASH_DEVICE_NAME);
+
+	if (!flash_dev) {
+		return OT_ERROR_NOT_IMPLEMENTED;
+	}
+
+	pages_count = flash_get_page_count(flash_dev);
+
+	if (flash_get_page_info_by_idx(flash_dev,
+		pages_count - CONFIG_OT_PLAT_FLASH_PAGES_COUNT - 1, &info)) {
```

Comment:
@KaSroka (for my understanding) what is the valid value range of CONFIG_OT_PLAT_FLASH_PAGES_COUNT? Note: the return value will not be checked.

Reply:
No less than 2 pages but also you cannot overwrite data in flash. Current implementation uses CONFIG_OT_PLAT_FLASH_PAGES_COUNT pages located at the end of FLASH so if this count is too big it will collide with program data.